developers/products/interchain-messaging-agent/enabling-tokens-on-schains.adoc

```diff
@@ -0,0 +1,126 @@
+= IMA Token Access Control
+:icons: font
+:toc: macro
+
+ifdef::env-github[]
+
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+
+endif::[]
+
+toc::[]
+
+== Introduction
+
+The SKALE Chain owner has special permissions to setup specific IMA parameters, such as IMA access control. These permissions include:
+
+* Enable/disable whitelist on mainnet (default: enabled)
+* Enable/disable automatic deployment on schain (default: disabled)
+* AddERC20TokenByOwner on mainnet and on schain
+* AddERC721TokenByOwner on mainnet and on schain
+
+== Enable or Disable the Whitelist
+
+The token whitelist allows the Schain owner to allow token exchange through IMA and a SKALE chain. For example, a decentralized exchange may prefer to disable the whitelist to allow any token transfer. A dApp using only one token may enable the whitelist to allow only a single token exchange. 
+
+To disable the whitelist, the owner executes `disableWhitelist` in LockAndDataForMainnetERC20 or LockAndDataForMainnetERC721 contracts, and pass in the schainName as the only argument. For example:
+
+```python
```

Comment:
is there a reason this is python? In other dev documents, we use js. Do you think it would be better to keep it consistent?  Especially, these codes have a `copy` option in the UI.

Reply:
I'll work on the js after the push